diff --git a/CRM/Mailing/BAO/Component.php b/CRM/Mailing/BAO/Component.php
index 2c23cdf..55dd8b3 100644
--- a/CRM/Mailing/BAO/Component.php
+++ b/CRM/Mailing/BAO/Component.php
@@ -98,7 +98,7 @@ class CRM_Mailing_BAO_Component extends CRM_Mailing_DAO_Component {
       $component->body_text = CRM_Utils_String::htmlToText(CRM_Utils_Array::value('body_html', $params));
     }
 
-    if ($component->is_default) {
+    if ($component->is_default && !empty($id)) {
       CRM_Core_DAO::executeQuery("UPDATE civicrm_mailing_component SET is_default = 0 WHERE component_type ='{$component->component_type}' AND id <> $id");
     }
 
