--- pear-db.orig/DB/mysqli.php	2025-01-29 20:58:51.120858808 -0800
+++ pear-db.new/DB/mysqli.php	2025-01-29 21:42:09.247526090 -0800
@@ -303,6 +303,8 @@
         {
             mysqli_report(MYSQLI_REPORT_OFF);
         }
+
+        $phpWarnings = '';
         if (((int) $this->getOption('ssl')) === 1) {
             $init = mysqli_init();
             mysqli_ssl_set(
@@ -313,7 +315,7 @@
                 empty($dsn['capath']) ? null : $dsn['capath'],
                 empty($dsn['cipher']) ? null : $dsn['cipher']
             );
-            if ($this->connection = @mysqli_real_connect(
+            if ($this->connection = $this->_mysqliRealConnect($phpWarnings,
                     $init,
                     $dsn['hostspec'],
                     $dsn['username'],
@@ -341,11 +343,11 @@
         if (!$this->connection) {
             if (($err = @mysqli_connect_error()) != '') {
                 return $this->raiseError(DB_ERROR_CONNECT_FAILED,
-                                         null, null, null,
+                                         null, null, $phpWarnings,
                                          $err);
             } else {
                 return $this->raiseError(DB_ERROR_CONNECT_FAILED,
-                                         null, null, null,
+                                         null, null, $phpWarnings,
                                          $php_errormsg);
             }
         }
@@ -358,6 +360,32 @@
     }
 
     // }}}
+    // {{{ _mysqliRealConnect()
+    /**
+     * Wrapper for mysqli_real_connect() which captures and returns any PHP warnings.
+     *
+     * @see \mysqli_real_connect()
+     */
+    function _mysqliRealConnect(&$warnings, $mysql, ?string $hostname, ?string $username, ?string $password, ?string $database, ?int $port, ?string $socket, int $flags = 0): bool {
+        set_error_handler(function ($errno, $errstr, $errfile, $errline) use (&$warnings) {
+            if ($errno & (E_WARNING|E_NOTICE|E_USER_WARNING)) {
+              if ($warnings === NULL) {
+                $warnings = '';
+              }
+              $warnings .= "\n$errstr";
+              return TRUE;
+            }
+            return FALSE;
+        });
+
+        try {
+            return mysqli_real_connect($mysql, $hostname, $username, $password, $database, $port, $socket, $flags);
+        }
+        finally {
+            restore_error_handler();
+        }
+    }
+    // }}}
     // {{{ disconnect()
 
     /**
