--- src/Common/XMLReader.php	2019-02-24 13:35:42.895637978 +1100
+++ src/Common/XMLReader.php	2019-02-24 11:41:29.462449569 +1100
@@ -71,10 +71,11 @@
      */
     public function getDomFromString($content)
     {
-        libxml_disable_entity_loader(true);
+        $originalLibXMLEntityValue = libxml_disable_entity_loader(true);
         $this->dom = new \DOMDocument();
         $this->dom->loadXML($content);
-
+        libxml_disable_entity_loader($originalLibXMLEntityValue);
+
         return $this->dom;
     }
 
