// karma/bin/karma.conf.js
module.exports = function(config) {
  config.set({

    // base path that will be used to resolve all patterns (eg. files, exclude)
    basePath: '../../../',


    // frameworks to use
    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
    frameworks: ['jasmine'],


    // list of files / patterns to load in the browser
    files: [
			 'packages/backbone/underscore-min.js',
       'packages/jquery/*.js',
       'packages/bower_components/angular/*.js',
			 'packages/bower_components/angular-mocks/angular-mocks.js',
			 'packages/backbone/*.js',
			 //'packages/backbone-forms/src/form.js',
       //'packages/backbone-forms/src/editors.js',
			 //'packages/backbone-forms/distribution/editors/jquery-ui.min.js',
			 'packages/jquery/plugins/jquery.civicrm-validate.js', 
			 'packages/jquery/plugins/select2/select2.min.js',
       'packages/bower_components/angular-route/*.js',
       'packages/bower_components/angular-ui-sortable/*.js',
       'packages/jquery/jquery-ui-1.9.0/development-bundle/ui/minified/*.js',

       'packages/bower_components/angular-ui-utils/ui-utils.js',
       'tests/mailing_test/controllerTest.js',

       'js/Common.js',
       'js/crm.ajax.js',


       'js/angular-newMailing.js'



    ],


    // list of files to exclude
    exclude: [
      
    ],


    // preprocess matching files before serving them to the browser
    // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
    preprocessors: {
    
    },


    // test results reporter to use
    // possible values: 'dots', 'progress'
    // available reporters: https://npmjs.org/browse/keyword/karma-reporter
    reporters: ['progress'],


    // web server port
    port: 9876,


    // enable / disable colors in the output (reporters and logs)
    colors: true,


    // level of logging
    // possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
    logLevel: config.LOG_INFO,


    // enable / disable watching file and executing tests whenever any file changes
    autoWatch: true,

    plugins : [
            'karma-chrome-launcher',
            'karma-jasmine'
            ],		
    // start these browsers
    // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
    browsers: ['Chrome'],


    // Continuous Integration mode
    // if true, Karma captures browsers, runs the tests and exits
    singleRun: false
  });
};
